indico/modules/categories/client/js/components/CategoryStatistics.jsx

```diff
@@ -0,0 +1,211 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import categoryStatisticsURL from 'indico-url:categories.statistics_json';
+
+import moment from 'moment';
+import React, {useCallback, useMemo} from 'react';
+import PropTypes from 'prop-types';
+import {Chart} from 'react-charts';
+import {Container, Header, Segment, Statistic, Message, Loader} from 'semantic-ui-react';
+import {Translate} from 'indico/react/i18n';
+
+import {useIndicoAxios} from 'indico/react/hooks';
+
+export default function CategoryStatistics({categoryId, lang}) {
+  const {data, loading: isLoading} = useIndicoAxios({
+    url: categoryStatisticsURL({category_id: categoryId}),
+    camelize: true,
+    trigger: categoryId,
+  });
+
+  if (isLoading) {
+    return <Loader size="massive" active />;
+  } else if (!data) {
+    return null;
+  }
+
+  return <CategoryStatisticsDisplay data={data} lang={lang} />;
+}
+
+CategoryStatistics.propTypes = {
+  categoryId: PropTypes.number.isRequired,
+  lang: PropTypes.string.isRequired,
+};
+
+function CategoryStatisticsDisplay({data, lang}) {
+  const userLanguage = lang.replace('_', '-');
+  const updatedTime = moment.utc(data.updated).format('D MMM YYYY HH:mm');
+  const fmt = new Intl.NumberFormat(userLanguage);
+  const totalEvents = fmt.format(data.totalEvents);
+  const totalContributions = fmt.format(data.totalContributions);
+  const files = fmt.format(data.files);
+  const users = fmt.format(data.users);
+  const serialize = useCallback(
+    statsData =>
+      Object.entries(statsData)
+        .filter(([year]) => year <= data.maxYear + 3 && year >= data.minYear)
+        .map(([year, number]) => [moment(year, 'YYYY'), number]),
+    [data.maxYear, data.minYear]
+  );
+
+  const eventGraph = useMemo(
+    () => [
+      {
+        label: Translate.string('Events'),
+        data: serialize(data.events),
+      },
+    ],
+    [data.events, serialize]
+  );
+  const contributionsGraph = useMemo(
+    () => [
+      {
+        label: Translate.string('Contributions'),
+        data: serialize(data.contributions),
+      },
+    ],
+    [data.contributions, serialize]
+  );
+  const series = useMemo(
```

Comment:
Does it have to be inside this component? Can't you move it outside of the function? The same applies to `axes` and `tooltip`

Reply:
I will give it a try to keep them outside and then calling it where it is used. What will be the difference if we keep it inside or outside?

Reply:
If something is completely static then you can indeed define it outside; that way it doesn't change during renders so no `useMemo` is needed. Since most of the values do depend on data though and thus need to be inside the component, I'm not sure how useful it is.

One option would be to create a single function outside that returns everything, and then only put something like this in the component:

    const {eventGraph, contribGrapy, ...} = useMemo(() => getGraphData(data), [...])

Reply:
Adrian was faster than me xD yeah, my main point was this is purely static and there is no need to even call `useMemo` on every render

Reply:
This is the reason for `useMemo` - https://github.com/tannerlinsley/react-charts#memoize-your-props

Reply:
Yeah, but you will achieve pretty much the same if you move the definitions of these props outside of the function. They won't change thus having the same effect as if you were using `useMemo`